tutorials/chapter1.md

```diff
@@ -0,0 +1,650 @@
+# Tutorial Contents
+
+## Pre-requisites
+
+* Medium level of C++ and basic understanding of C++11 [share pointers](https://en.cppreference.com/w/cpp/memory/shared_ptr) and [lambda functions](https://en.cppreference.com/w/cpp/language/lambda).
+* Understanding of the publisher/subscriber pattern [example](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern)
+* Basic knowledge of [CMAKE](https://cmake.org/)
+* [Unit testing](https://en.wikipedia.org/wiki/Unit_testing) knowledge and [google test](https://github.com/google/googletest) framework
+* Although not required beforehand, there will be use of [YAML language](https://yaml.org/).
+* Klepsydra recommends the use of the [composition root](https://medium.com/@cfryerdev/dependency-injection-composition-root-418a1bb19130) and [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection) patterns. These two concepts appear repeatedly throughout this tutorial.
+
+For the ROS tutorial, basic knowledge of ROS is required and similarly for the DDS tutorial, RTI DDS understanding is required.
+
+## Technical requirements
+
+Installation of Klepsydra Core, Optionally ROS, DDS.
+
+Klepsydra is installed (by default) to /usr/local by CMake. To develop an application using Klepsydra use the cmake command `find_package(Klepsydra)` in the CMakeLists file.. With this command, the Klepsydra includes are available in the variable `KLEPSYDRA_INCLUDE_DIRS` and the libraries are available using the variable `KLEPSYDRA_CORE_LIBRARIES`. The CMakeLists file provided for compiling the examples in the tutorial demonstrate how these two variables are used.
+
+## Chapter 1
+
+
+### Introduction
+
+Klepsydra API is split into main groups:
+
+* **Application API**. A simple Publisher / Subscriber API, this API is used by the application or customer code to build the functionality.
+* **Composition API**. This API is used to create and assembly publisher subscriber instances. Examples are: EventLoop, ROS Connector, etc.
```

Comment:
How does this affect the structure of the tutorial? I guess **Application API** is somehow simpler than **Composition API** Are we going to cover both?

Reply:
Both are fully covered in chapter 1. Let me know if this is not clear?